### PR TITLE
Fix build when `providerless` tags is provided

### LINKED ISSUE #85941: Compiling the kubelet in providerless mode is broken for release-1.14 & release-1.15
**What happened**: Compiling the kubelet in providerless mode fails with

```shell
# k8s.io/kubernetes/pkg/cloudprovider/providers/azure
pkg/cloudprovider/providers/azure/azure_controller_common.go:66:13: undefined: lockMap
```

This is the issue for both
- [release-1.14](https://hush-house.pivotal.io/teams/k8s-c10s/pipelines/kind/jobs/kind%20master%20with%20k8s-current-minus-2/builds/259)
- [release-1.15](https://hush-house.pivotal.io/teams/k8s-c10s/pipelines/kind/jobs/kind%20master%20with%20k8s-current-minus-1/builds/254)

**What you expected to happen**: No compilation error

**How to reproduce it (as minimally and precisely as possible)**:

```shell
mkdir /tmp/lockless && cd /tmp/lockless
git clone https://github.com/kubernetes/kubernetes && cd kubernetes
GOFLAGS=-tags=providerless KUBE_VERBOSE=0 KUBE_BUILD_PLATFORMS=linux/amd64 ./build/run.sh make all WHAT=cmd/kubelet
# ==> fails
```

**Anything else we need to know?**:

Those are most probably the CPs that have introduced the issue:
- release-1.14: https://github.com/kubernetes/kubernetes/pull/85832
- release-1.15: https://github.com/kubernetes/kubernetes/pull/85830

[release-1.16](https://hush-house.pivotal.io/teams/k8s-c10s/pipelines/kind/jobs/kind%20master%20with%20k8s-current/builds/255) & [master](https://hush-house.pivotal.io/teams/k8s-c10s/pipelines/kind/jobs/kind%20master%20with%20k8s-master/builds/262) seem to be fine.

The original PR which has been cherry-picked into the releases is https://github.com/kubernetes/kubernetes/pull/85115.

/assign @aramase @andyzhangx
/kind bug
/sig cloud-provider
/area provider/azure

**Environment**:
- Kubernetes version (use `kubectl version`): release-1.1{4,5}
- Cloud provider or hardware configuration: providerless
- OS (e.g: `cat /etc/os-release`): (compiling on) Darwin (via docker) and Linux
- Kernel (e.g. `uname -a`): -
- Install tools: -
- Network plugin and version (if this is a network-related bug): -
- Others: -

### PR DESCRIPTION
**Pull request by hoegaarden**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**: 

If someone tries to compile int providerless mode / with the build tag
'providerless', some bits are excluded and compilation will fail. This
is due to the fact that in this release we don't support providerless
yet.
Therefore we should remove the build tags from the files to not
accidentaly exclude them.

**Which issue(s) this PR fixes**:

Fixes #85941

**Special notes for your reviewer**:

This is not a automated CP, as this is a specific fix to this branch and not a fix we want in master.

The same fix has been PR'ed into release-1.15: https://github.com/kubernetes/kubernetes/pull/85946

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/sig cloud-provider
/area provider/azure
/assign @aramase @andyzhangx 
/milestone v1.14
/priority important-soon

### CODE REVIEW

**Comment by andyzhangx** [comment]
/hold
@hoegaarden this is a cherry-pick fix, we should first fix it in master branch, and then decide whether to cherry-pick, and could use cherry-pick tool to port the fix into old releases.

**Comment by andyzhangx** [comment]
/hold cancel
I was wrong, we should fix this in 1.14 & 1.15

**Comment by hoegaarden** [comment]
> @hoegaarden this is a cherry-pick fix, we should first fix it in master branch, and then decide whether to cherry-pick, and could use cherry-pick tool to port the fix into old releases.

We don't want that in `master`. This is a fix for the release branch only. Therefore I opened the PR to the branch directly.

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Comment by justaugustus** [comment]
Release Manager + Azure approval:
/lgtm
/approve

**Comment by liggitt** [comment]
why was this merged? this appears to go the wrong direction, by explicitly including things in the providerless build that should not be there?

cc @BenTheElder

**Comment by liggitt** [comment]
> This is due to the fact that in this release we don't support providerless yet.

If we don't support it, why are we merging fixes related to it?

**Comment by justaugustus** [comment]
> why was this merged? this appears to go the wrong direction, by explicitly including things in the providerless build that should not be there?

@liggitt -- As I understand it, providerless was not added until 1.16, so the build tags that were part of https://github.com/kubernetes/kubernetes/pull/85115, should not have been introduced to this branch.

All the same, your comment on https://github.com/kubernetes/kubernetes/issues/85941#issuecomment-562147584:
> if the providerless tag is not supported in 1.14/1.15, I would not expect to be able to use it with builds in those branches

is completely accurate.

**Pull request merged**
